src/main/java/org/cactoos/http/HtWire.java

```diff
@@ -80,16 +86,40 @@ public HtWire(final String addr) {
      * @param tcp The TCP port
      */
     public HtWire(final String addr, final int tcp) {
+        this(addr, tcp, HtWire::socket);
+    }
+
+    /**
+     * Ctor.
+     * @param uri The address of the server
+     * @param spplier Socket supplier
+     */
+    HtWire(final URI uri,
+        final BiFunction<String, Integer, Socket> spplier) {
+        this(uri.getHost(), HtWire.prt(uri), spplier);
```

Comment:
@krzyk  Another rule of design principles says "No code in constructors" which means you can only store arguments into fields (or create other objects). But here you call to `static` method `prt` in the constructor.